ingest-app/src/cmr/ingest/api/provider.clj

```diff
@@ -1,38 +1,130 @@
 (ns cmr.ingest.api.provider
-  "Defines the HTTP URL routes for the provider endpoint in the ingest application."
+  "Defines the HTTP URL routes for the provider endpoint in the ingest application.
+
+   Example usage:
+   server=\"http://localhost:3002\"
+   token=\"mock-echo-system-token\"
+   file_path=\"schemas/resources/schemas/provider/v1.0.0/metadata.json\"
+   curl -s \\
+       -X PUT \\
+       -H \"Cmr-Pretty: true\" \\
+       -H \"Content-type: application/json\" \\
+       -H \"Authorization: $token\" \\
+       $server/providers/PROVX \\
+       -d \"@$full_path\"
+   "
   (:require
    [cheshire.core :as json]
+   [clojure.string :as string]
    [cmr.acl.core :as acl]
    [cmr.common-app.api.enabled :as common-enabled]
+   [cmr.common.generics :as gcom]
    [cmr.common.mime-types :as mt]
    [cmr.common.services.errors :as srvc-errors]
-   [cmr.ingest.services.provider-service :as ps]
-   [compojure.core :refer :all]))
+   [cmr.ingest.services.provider-service :as provider-service]
+   [compojure.core :refer :all]
+   [cmr.common.generics :as generics]
+   [cmr.common.services.errors :as errors]
+   [cmr.common.util :as util]))
+
+(defn- drop-metadata
+  "Look for and remove metadata field from the body returned by the metadata db
+   service, as it is not needed in the legacy responses."
+  [response]
+  (let [body (-> response
+                 :body
+                 json/parse-string
+                 (as-> metadata (map #(dissoc %1 "metadata") metadata))
```

Comment:
I believe that when there is only 1 param for the mapping function, it is idiomatic to not use numbers eg. `%` instead of `%1`